xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-webjar/src/main/webjar/lightbox.js

```diff
@@ -0,0 +1,241 @@
+      $('.lightboxDescription').css('display', 'block');
+      $('.lightboxDescription').find('.title').text(data.name);
+      var translationPublisher = l10n.get('author', XWiki.Model.resolve(data.author, XWiki.EntityType.DOCUMENT).name);
+      $('.lightboxDescription').find('.publisher').text(translationPublisher);
+      var translationDate = l10n.get('date', new Date(data.date).toLocaleDateString());
+      $('.lightboxDescription').find('.date').text(translationDate);
+
+      if (caption != undefined) {
+        $('.lightboxDescription').find('.caption').text(caption);
+      } else {
+        $('.lightboxDescription').find('.caption').empty();
+      }
+    } else {
+      $('.lightboxDescription').css('display', 'none');
+    }
+  };
+
+  /**
+   * Get information for attachment, when this can be found inside the current page.
+   * TODO: consider the path of the attachment, not just the current page.
+   */
+  var getAttachmentInfo = function(imageLink) {
+    var deferred = $.Deferred();
+    var attachmentsURL = xm.restURL + `/attachments`;
+    $.ajax(attachmentsURL, {
+      method: 'GET',
+      dataType: 'json'
+    }).done(function (data, textStatus, jqXHR) {
+      var attachment = data.attachments.find(field => field.name == getImageName(imageLink));
+      deferred.resolve(attachment);
+    }).fail(function () {
+      deferred.reject();
+    });
+    return deferred.promise();
+  };
+
+  var getImageCaption = function(image) {
+    if (image.closest('figure').length > 0) {
+      return image.closest('p').siblings('figcaption').text();
+    }
+  };
+
+  /**
+   * Open Gallery lightbox at the current index.
+   */
+  var openLightbox = function() {
+    var media = [];
+    $('#xwikicontent').find('img').each(function(index) {
+      var imageURL = removeResizeParams(this.src);
+      var caption = getImageCaption($(this));
+      media.push({
+        href: imageURL,
+        thumbnail: createThumbnailURL(imageURL),
+        caption: caption
+      });
+    });
+
+    var options = {
+      container: '#blueimp-gallery',
+      index: parseInt($('#openLightbox').data('index')),
+      // The class names are overridden since we changed the styles.
+      closeClass: 'escape',
+      playPauseClass: 'autoPlay',
+      onslide: function(index, slide) {
+        var caption = this.list[index].caption;
+        getAttachmentInfo($(slide).find('img')[0].src).done(function(data) {
+          updateLightboxDescription(data, caption);
+        });
+      }
+    };
+    myOpenLightbox = gallery(media, options);
+  };
+
+  /**
+   * Initialize the lightbox functionality for a set of images.
+   */
+  var initLightboxFunctionality = function() {
+    console.log("Init");
+    enableToolbarPopovers();
+  };
+
+  $(document).on('click', '#openLightbox', openLightbox);
+
+  $(document).on('click', '#lightboxFullscreen', toggleLightboxFullscreen);
+
+  $(window).on('load', function() {
+    initLightboxFunctionality();
+  });
```

Comment:
```suggestion
  $(function() {
    initLightboxFunctionality();
  });
```

The ``load`` event can easily be fired before your JavaScript code is executed (it happened for me on the first try of the lightbox) so don't rely on that. Just use https://api.jquery.com/ready/#ready-handler with its recommended form ``$(handler)``.